run_service.py

```diff
@@ -0,0 +1,228 @@
+                "fund_requirements": FundRequirementsTemplate(
+                    {
+                        "agent": SUGGESTED_TOP_UP_DEFAULT,
+                        "safe": SUGGESTED_SAFE_TOP_UP_DEFAULT,
+                    }
+                ),
+            }
+        ),
+    }
+)
+
+
+def print_box(text: str, margin: int = 1, character: str = '=') -> None:
+    """Print text centered within a box."""
+
+    lines = text.split('\n')
+    text_length = max(len(line) for line in lines)
+    length = text_length + 2 * margin
+
+    border = character * length
+    margin_str = ' ' * margin
+
+    print(border)
+    print(f"{margin_str}{text}{margin_str}")
+    print(border)
+    print()
+
+
+def print_title(text: str) -> None:
+    """Print title."""
+    print()
+    print_box(text, 4, '=')
+
+
+def print_section(text: str) -> None:
+    """Print section."""
+    print_box(text, 1, '-')
+
+
+def wei_to_unit(wei: int) -> float:
+    """Convert Wei to unit."""
+    return wei / 1e18
+
+
+def wei_to_token(wei: int, token: str = "xDAI") -> str:
+    """Convert Wei to token."""
+    return f"{wei_to_unit(wei):.2f} {token}"
+
+
+def ask_confirm_password() -> str:
+    password = getpass.getpass("Please enter a password: ")
+    confirm_password = getpass.getpass("Please confirm your password: ")
+
+    if password == confirm_password:
+        return password
+    else:
+        print("Passwords do not match. Terminating.")
+        sys.exit(1)
+
+
+def check_rpc(rpc_url: str) -> None:
+    spinner = Halo(text=f"Checking RPC...", spinner="dots")
+    spinner.start()
+
+    rpc_data = {
+        "jsonrpc": "2.0",
+        "method": "eth_newFilter",
+        "params": ["invalid"],
+        "id": 1
+    }
+
+    try:
+        response = requests.post(
+            rpc_url,
+            json=rpc_data,
+            headers={"Content-Type": "application/json"}
+        )
+        response.raise_for_status()
+        rpc_response = response.json()
+    except Exception as e:
+        print("Error: Failed to send RPC request:", e)
+        sys.exit(1)
+
+    rcp_error_message = rpc_response.get("error", {}).get("message", "Exception processing RCP response")
+
+    if rcp_error_message == "Exception processing RCP response":
+        print("Error: The received RCP response is malformed. Please verify the RPC address and/or RCP behavior.")
+        print("  Received response:")
+        print("  ", rpc_response)
+        print("")
+        print("Terminating script.")
+        sys.exit(1)
+    elif rcp_error_message == "Out of requests":
+        print("Error: The provided RCP is out of requests.")
+        print("Terminating script.")
+        sys.exit(1)
+    elif rcp_error_message == "The method eth_newFilter does not exist/is not available":
+        print("Error: The provided RPC does not support 'eth_newFilter'.")
+        print("Terminating script.")
+        sys.exit(1)
+    
+    spinner.succeed("RPC checks passed.")
+
+
+def main() -> None:
```

Comment:
is it possible to have a couple of flags here;

A) --dry-run: (effectively validate prior to the the deployment to verify that the deployment will proceed without issue?

B) --local-user-account-password/--no-user-password:  Flag such that by default none is passed, and it will prompt the user, however if it is passed this will be used.

Reply:
C) --wait-for balance

Reply:
Yes, we already have some of these features in main/develop branch. Will comment to the team about the rest.